File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/C++/s221078931_comments_removed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

class IntegralRect {
public:
    int height;
    int width;

    IntegralRect(int h, int w) : height(h), width(w) {}

    int diagonal() const {
        if (height >= width) {
            return std::numeric_limits<int>::max();
        }
        return height * height + width * width;
    }

    bool operator<(const IntegralRect& o) const {
        int d1 = diagonal();
        int d2 = o.diagonal();
        return (d1 == d2) ? height < o.height : d1 < d2;
    }
};

int main() {
    std::vector<IntegralRect> rects;
    for (int i = 0; i < 200; i++) {
        for (int j = 0; j < 200; j++) {
            rects.emplace_back(i + 1, j + 1);
        }
    }
    std::sort(rects.begin(), rects.end());

    int h, w;
    while (std::cin >> h >> w) {
        if (h == 0 && w == 0) {
            break;
        }

        IntegralRect tmpRect(h, w);
        auto it = std::upper_bound(rects.begin(), rects.end(), tmpRect);

        if (it != rects.end()) {
            std::cout << it->height << " " << it->width << std::endl;
        }
    }

    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Go/s221078931_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

type IntegralRect struct {
	height int
	width  int
}

func (r IntegralRect) diagonal() int {
	if r.height >= r.width {
		return int(^uint(0) >> 1) // Max int value
	}
	return r.height*r.height + r.width*r.width
}

func (r IntegralRect) compareTo(o IntegralRect) int {
	d1 := r.diagonal()
	d2 := o.diagonal()

	if d1 == d2 {
		return r.height - o.height
	}
	return d1 - d2
}

type IntegralRects []IntegralRect

func (r IntegralRects) Len() int           { return len(r) }
func (r IntegralRects) Less(i, j int) bool { return r[i].compareTo(r[j]) < 0 }
func (r IntegralRects) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }

func main() {
	var rects []IntegralRect
	for i := 1; i <= 200; i++ {
		for j := 1; j <= 200; j++ {
			rects = append(rects, IntegralRect{height: i, width: j})
		}
	}
	sort.Sort(IntegralRects(rects))

	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		input := scanner.Text()
		tmpArray := strings.Fields(input)
		h, _ := strconv.Atoi(tmpArray[0])
		w, _ := strconv.Atoi(tmpArray[1])

		if h == 0 && w == 0 {
			break
		}

		tmpRect := IntegralRect{height: h, width: w}
		diag := tmpRect.diagonal()

		for i := 0; i < len(rects); i++ {
			if rects[i].compareTo(tmpRect) > 0 {
				fmt.Printf("%d %d\n", rects[i].height, rects[i].width)
				break
			}
		}
	}
}

// End of Code
